### PR TITLE
Migrate and adapt the wazuh agent subsystem

### PR DESCRIPTION
**Pull request by lchico**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/22080|

## Description
As the main task of this PR is to move all required components from the wazuh-package repository to be able to generate RPM, DEB, ZIP, and PKG packages for the agent, the following steps were taken:

- Moved the wazuh-agent specs from the wazuh-packages repository to the wazuh repository.
- Unified the build scripts for DEB and RPM packages.
  - Utilized a helper function to define specific behavior for the generation of each package.
- Also unified the generate_package.sh script.
  - Defined the following environment variables:
    + `BUILD_TARGET`
    + `ARCHITECTURE_TARGET`
    + `PACKAGE_FORMAT`
    + `INSTALLATION_PATH`
- Used the sources flag to specify that the specs and source code will be taken from the wazuh-local-src directory.
- Update the package name to adhere to [this naming convention](https://github.com/wazuh/wazuh/issues/21755#:~:text=As%20subsystem%20owners%2C%20teams%20must%20follow%20this%20naming%20convention%20for%20development%20package%20names%3A%20%3CSubsystemName%3E_%3CVersionNumber%3E%2D%3CRevision%3E_%7B%3COSName%3E%2D%3COSVersion%3E_%7D%3CArchitecture%3E_%3CGitReference%3E.%3CPackageType%3E.) for development package names:
```
 <SubsystemName>_<VersionNumber>-<Revision>_{<OSName>-<OSVersion>_}<Architecture>_<GitReference>.<PackageType>.
```

## Generate package
To generate a DEB/RPM package for wazuh-agent:

```
git clone --branch 22080-migrate-and-adapt-the-wazuh-agent-subsystem https://github.com/wazuh/wazuh.git
cd wazuh/packages

# ./generate_package.sh -s /tmp -t agent -a [arch] -r [revision] --package-format [deb|rpm] 
package#  ./generate_package.sh -s /tmp -t agent -a amd64 -r test --package-format deb 
```

## Tests

| Reference                                                                                   | Architecture | System | Legacy | Is Stage | Status          |
|--------------------------------------------------------------------------------------------|--------------|--------|--------|----------|-----------------|
| [wazuh-agent_4.9.0-lctest_amd64_f1aa6c8.deb](https://github.com/wazuh/wazuh/actions/runs/8621711487)             | amd64        | deb    | false  | false    | :green_circle: |
| [wazuh-agent_4.9.0-lctest_amd64_f1aa6c8.deb](https://github.com/wazuh/wazuh/actions/runs/8621711578)             | amd64        | deb    | false  | false    | :green_circle: |
| [wazuh-agent_4.9.0-lctest_i386_f1aa6c8.deb](https://github.com/wazuh/wazuh/actions/runs/8621711774)             | i386         | deb    | false  | false    | :green_circle: |
| [wazuh-agent-4.9.0-lctest.i386.rpm](https://github.com/wazuh/wazuh/actions/runs/8621711897)                       | i386         | rpm    | false  | true    | :green_circle: |
| [wazuh-agent_4.9.0-lctest_i386_f1aa6c8.deb](https://github.com/wazuh/wazuh/actions/runs/8621712015)             | i386         | deb    | false  | false    | :green_circle: |
| [wazuh-agent-4.9.0-lctest.i386.rpm](https://github.com/wazuh/wazuh/actions/runs/8621712103)                       | i386         | rpm    | false  | true    | :green_circle: |
| [wazuh-agent_4.9.0-lctest.el5_i386_f1aa6c85ee4.rpm](https://github.com/wazuh/wazuh/actions/runs/8621712147)      | i386         | rpm    | true   | false    | :green_circle: |
| [wazuh-agent_4.9.0-lctest.el5_i386_f1aa6c85ee4.rpm](https://github.com/wazuh/wazuh/actions/runs/8621712296)      | i386         | rpm    | true   | false    | :green_circle: |
| [wazuh-agent-4.9.0-lctest.x86_64.rpm](https://github.com/wazuh/wazuh/actions/runs/8621712382)                     | x86_64       | rpm    | false  | true    | :green_circle: |
| [wazuh-agent-4.9.0-lctest.x86_64.rpm](https://github.com/wazuh/wazuh/actions/runs/8621712573)                     | x86_64       | rpm    | false  | true    | :green_circle: |
| [wazuh-agent_4.9.0-lctest.el5_x86_64_f1aa6c85ee4.rpm](https://github.com/wazuh/wazuh/actions/runs/8621712628)    | x86_64       | rpm    | true   | false    | :green_circle: |
| [wazuh-agent_4.9.0-lctest.el5_x86_64_f1aa6c85ee4.rpm](https://github.com/wazuh/wazuh/actions/runs/8621712821)    | x86_64       | rpm    | true   | false    | :green_circle: |

- Compilation without warnings in every supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [x] Package installation

### CODE REVIEW

File: .github/workflows/rtr_syscheck.yml [hunk 1]
```diff
@@ -9,7 +9,6 @@ on:
       - "src/unit_tests/syscheckd/**"
       - "src/Makefile"
       - ".github/workflows/rtr_syscheck.yml"
-      - ".github/actions/**"
```
Comment (jotacarma90) [inline .github/workflows/rtr_syscheck.yml:12]: We should remove this changes, or at least improve the filtering of these workflows.

File: packages/debs/SPECS/wazuh-agent/debian/conffiles
Comment (jotacarma90) [inline packages/debs/SPECS/wazuh-agent/debian/conffiles:1]: Is this file needed?

File: packages/macos/entitlements.plist [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <key>com.apple.security.files.downloads.read-only</key>
+    <true/>
+    <key>com.apple.security.assets.pictures.read-only</key>
+    <true/>
+    <key>com.apple.security.assets.music.read-only</key>
+    <true/>
+    <key>com.apple.security.assets.movies.read-only</key>
+    <true/>
+  </dict>
+</plist>
```
Comment (jotacarma90) [inline packages/macos/entitlements.plist:22]: Add new line at the end of file

File: packages/macos/generate_wazuh_packages.sh [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+#
+# This program is a free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+CURRENT_PATH="$( cd $(dirname ${0}) ; pwd -P )"
+ARCH="intel64"
+WAZUH_SOURCE_REPOSITORY="https://github.com/wazuh/wazuh"
+INSTALLATION_PATH="/Library/Ossec"    # Installation path
```
Comment (jotacarma90) [inline packages/macos/generate_wazuh_packages.sh:15]:
```suggestion
INSTALLATION_PATH="/Library/Ossec"    # Installation path.
```

File: packages/macos/generate_wazuh_packages.sh [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+VERSION=""                            # Default VERSION (branch/tag)
```
Comment (jotacarma90) [inline packages/macos/generate_wazuh_packages.sh:16]:
```suggestion
VERSION=""                            # Default VERSION (branch/tag).
```

File: packages/macos/generate_wazuh_packages.sh [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+REVISION="1"                          # Package revision.
+BRANCH_TAG=""                         # Branch that will be downloaded to build package.
+DESTINATION="${CURRENT_PATH}/output/" # Where package will be stored.
+JOBS="2"                              # Compilation jobs.
+VERBOSE="no"                          # Enables the full log by using `set -exf`.
+DEBUG="no"                            # Enables debug symbols while compiling.
+CHECKSUMDIR=""                        # Directory to store the checksum of the package.
+CHECKSUM="no"                         # Enables the checksum generation.
+IS_STAGE="no"                         # Enables release package naming,
```
Comment (jotacarma90) [inline packages/macos/generate_wazuh_packages.sh:25]:
```suggestion
IS_STAGE="no"                         # Enables release package naming.
```

File: packages/macos/package_files/preinstall.sh [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    sudo ${DSCL} localhost -createprop /Local/Default/Users/wazuh UserShell /usr/bin/false
+    sudo ${DSCL} localhost -createprop /Local/Default/Users/wazuh NFSHomeDirectory /var/wazuh
+    sudo ${DSCL} localhost -createprop /Local/Default/Users/wazuh UniqueID ${new_uid}
+    sudo ${DSCL} localhost -createprop /Local/Default/Users/wazuh PrimaryGroupID ${new_gid}
+    sudo ${DSCL} localhost -append /Local/Default/Groups/wazuh GroupMembership wazuh
+    sudo ${DSCL} localhost -createprop /Local/Default/Users/wazuh Password "*"
+fi
+
+#Hide the fixed users
+dscl . create /Users/wazuh IsHidden 1
```
Comment (jotacarma90) [inline packages/macos/package_files/preinstall.sh:143]: Add new line at the end of file

File: packages/rpms/amd64/agent/CentOS-Base.repo [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+gpgcheck=1
+enabled=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
+
+# SCLO - packages
+[centos-sclo-sclo]
+name=CentOS-$releasever - SCLO
+baseurl=http://mirror.nsc.liu.se/centos-store/6.10/sclo/$basearch/rh/
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
```
Comment (jotacarma90) [inline packages/rpms/amd64/agent/CentOS-Base.repo:54]: New line

File: packages/rpms/amd64/legacy/CentOS-Base.repo [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+gpgcheck=1
+enabled=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-5
+
+[fasttrack]
+name=CentOS-$releasever - CR
+baseurl=http://linuxsoft.cern.ch/centos-vault/5.11/fasttrack/$basearch/
+gpgcheck=1
+enabled=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-5
```
Comment (jotacarma90) [inline packages/rpms/amd64/legacy/CentOS-Base.repo:67]: New line

File: packages/rpms/i386/agent/CentOS-Base.repo [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+enabled=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
+
+#contrib - packages by Centos Users
+[contrib]
+name=CentOS-$releasever - Contrib
+baseurl=http://mirror.nsc.liu.se/centos-store/6.10/contrib/i386/
+gpgcheck=1
+enabled=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
```
Comment (jotacarma90) [inline packages/rpms/i386/agent/CentOS-Base.repo:47]: New line

File: packages/rpms/i386/legacy/CentOS-Base.repo [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+gpgcheck=1
+enabled=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-5
+
+[fasttrack]
+name=CentOS-$releasever - CR
+baseurl=http://linuxsoft.cern.ch/centos-vault/5.11/fasttrack/$basearch/
+gpgcheck=1
+enabled=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-5
```
Comment (jotacarma90) [inline packages/rpms/i386/legacy/CentOS-Base.repo:67]: New line

**Review by jotacarma90 — CHANGES_REQUESTED** [review]

File: packages/debs/SPECS/wazuh-agent/debian/conffiles
Comment (lchico) [reply packages/debs/SPECS/wazuh-agent/debian/conffiles:1]: While not mandatory, using debian/conffiles is considered a good practice for Debian package development. It enhances user experience by providing control over configuration changes during package installation and upgrades.

**Comment by lchico** [comment]
@jotacarma90 Thank you for your comments. All of them have been addressed. When you have a moment, PTAL. Thank you!

**Review by mjcr99 — CHANGES_REQUESTED** [review]
Amazing job @lchico! 
Just a comment, the upload images workflows should be discussed with the team.

**Review by mjcr99 — APPROVED** [review]
LGTM

**Review by jotacarma90 — APPROVED** [review]
LGTM

**Review by MarcelKemp — APPROVED** [review]
LGTM.

**Pull request merged**